tests/test_get_parser_unit.py

```diff
@@ -0,0 +1,174 @@
+import os
+import sys
+import time
+import pytest
+
+#
+# we assume everwhere our current directory is in the package 
+# test area, so go ahead and cd there
+#
+os.chdir(os.path.dirname(__file__))
+
+
+#
+# import modules we need to test, since we chdir()ed, can use relative path
+#
+sys.path.append("../lib")
+import get_parser
+
+from test_unit import TestUnit 
+
+@pytest.fixture
+def find_all_arguments():
+    # try to extract all the --foo arguments from the source
+    # and track which ones are flags
+    f = open("../lib/get_parser.py","r")
+    flagargs = set()
+    listargs = set()
+    allargs = []
+    dest = {}
+    for line in f.readlines():
+        p = line.find('"--')
+        if p > 0:
+            arg = line[p+3:]
+            arg = arg[0:arg.find('"')]
+            allargs.append(arg)
+            dest[arg] = arg
+        if line.find('dest="') > 0:
```

Comment:
For all of these `if` clauses finding pieces of the argument specification, what happens if someone specifies arguments using single-quotes?  Hopefully that'll go away once we implement black, but it is still valid python.  i.e. 

`parser.add_argument('--my-awesome-flag', dest='blahblah')`

Reply:
Yes, that's a possible problem.  I could check for both kinds of quotes... it shouldn't cost much.

Reply:
Oof.  If we  allow the other quotes everywhere, then we trip over help messages with options:
```
    parser.add_argument(
        "--OS",
        default=None,
        help="specify OS version of worker node. Example --OS=SL5 Comma seperated list '--OS=SL4,SL5,SL6' works as well . Default is any available OS",
    )
```
finds the '--OS=SL4,SL5,SL6...' in the help message, and tries to make it an option named OS=SL4,SL5,SL6... I think I can work around it though.

Okay, that works either way now.